### ISSUE TITLE
Feed Me getAllTicketsByEventId() Import Error

### ISSUE THREAD

**Issue by johnj24** [verbb/Events#58]
### Description
I'm importing events from Solspace Calendar, which never had any ticketing options before. The error I'm getting is as follows:

`Argument 1 passed to verbb\events\services\Tickets::getAllTicketsByEventId() must be of the type integer, string given, called in /storage/av04383/www/stage_html/vendor/verbb/events/src/elements/Event.php on line 370 - Tickets.php: 26`

### Steps to reproduce

1. Export Solspace Calendar Events using the Craft built-in element exporter on the Events index page.
2. Map the fields, leaving the ticketing fields with just their defaults, and run the import.

### Additional info

- Plugin version: 1.4.7
- Craft version: 3.4.13

**Comment by engram-design** [comment]
How are you getting this error, by running the debug action, or just running the feed? I can't seem to replicate it on my end. I'm using the following JSON:
```
{
    "events": {
        "event": [
            {
                "title": "Event Title",
                "startDate": "2017-05-11 07:00:00",
                "endDate": "2017-05-25 13:00:00",
                "allDay": 0
            },
            {
                "title": "Event Title 2",
                "startDate": "2017-05-11 07:00:00",
                "endDate": "2017-05-25 13:00:00",
                "allDay": 0
            },
            {
                "title": "Event Title 3",
                "startDate": "2017-05-11 07:00:00",
                "endDate": "2017-05-25 13:00:00",
                "allDay": 0
            }
        ]
    }
}
```

And my settings:

![Screen Shot 2020-04-21 at 12 45 56 pm](https://user-images.githubusercontent.com/1221575/79819790-15c5af00-83ce-11ea-9987-509c7b0348b5.png)

**Comment by johnj24** [comment]
My JSON is what Craft produced from it's exporter. I used the expanded setting. Here's a snippet with some example text thrown in.
```
[
    {
        "calendarId": "1",
        "name": "Races",
        "authorId": "1",
        "startDate": {
            "date": "2020-11-26 00:00:00.000000",
            "timezone_type": 2,
            "timezone": "UTC"
        },
        "initialStartDate": {
            "date": "2020-11-26 00:00:00.000000",
            "timezone_type": 2,
            "timezone": "UTC"
        },
        "startDateLocalized": {
            "date": "2020-11-26 00:00:00.000000",
            "timezone_type": 3,
            "timezone": "America/Chicago"
        },
        "endDate": {
            "date": "2020-11-29 23:59:59.000000",
            "timezone_type": 2,
            "timezone": "UTC"
        },
        "initialEndDate": {
            "date": "2020-11-29 23:59:59.000000",
            "timezone_type": 2,
            "timezone": "UTC"
        },
        "endDateLocalized": {
            "date": "2020-11-29 23:59:59.000000",
            "timezone_type": 3,
            "timezone": "America/Chicago"
        },
        "allDay": "1",
        "rrule": null,
        "freq": null,
        "interval": null,
        "count": null,
        "until": null,
        "untilLocalized": null,
        "byMonth": null,
        "byYearDay": null,
        "byMonthDay": null,
        "byDay": null,
        "sortOrder": null,
        "score": null,
        "username": "example",
        "id": "811403",
        "tempId": null,
        "draftId": null,
        "revisionId": null,
        "uid": "aa500e0e-0424-4b91-8524-209cb2cfad5a",
        "fieldLayoutId": "621",
        "contentId": "17499",
        "enabled": "1",
        "archived": "0",
        "siteId": "1",
        "title": "2020 Example Event Title",
        "slug": "2020-example-event-slug",
        "uri": "event/2020-example-event-slug",
        "dateCreated": "2019-09-30T15:47:04-05:00",
        "dateUpdated": "2020-04-11T15:01:55-05:00",
        "dateDeleted": null,
        "trashed": false,
        "propagateAll": false,
        "newSiteIds": [],
        "resaving": false,
        "duplicateOf": null,
        "previewing": false,
        "hardDelete": false,
        "ref": null,
        "status": "enabled",
        "structureId": null,
        "url": "https://example.com/event/2020-example-event-slug",
        "eventName": "Example Event Name",
        "coverImage": [],
        "eventLocation": [
            "6496"
        ],
        "addressName": null,
        "address": {
            "elementId": 811403,
            "fieldId": 92,
            "handle": "address",
            "street1": null,
            "street2": null,
            "city": null,
            "state": null,
            "zip": null,
            "country": null,
            "lat": null,
            "lng": null,
            "coords": null,
            "distance": null
        },
        "eventLogo": [],
        "eventSeries": [],
        "eventOrganizer": [
            "206880"
        ],
    }
]
```
Should I have the root elements `{ events: { event: [ ... ] } }` ?

Here are some of the settings: 
![admin_feed-me_feeds_map_32 (1)](https://user-images.githubusercontent.com/1458917/79884116-5ca9a980-83ba-11ea-9e73-a954626d53e1.png)

**Comment by johnj24** [comment]
Here's another error I get that has to do (I think) with leaving the Ticket Price default value empty.
`A non well formed numeric value encountered - Tickets.php: 26`

![admin_feed-me_logs](https://user-images.githubusercontent.com/1458917/79915051-e02fbe80-83eb-11ea-955a-dcc830f0d1f7.png)

**Comment by engram-design** [comment]
No need to have the extra levels for the JSON - using the root element is perfectly fine.

I notice you're mapping the Event ID to the `uid` - this will be an issue. You shouldn't ever map this unless you want to update an existing event element specifically. It must match the event element ID exactly.

I've just realised I've left `Solspace` text in that mapping description, which is incorrect. It should be for the Verbb Event ID. Fixed in [1.4.8](https://github.com/verbb/events/releases/tag/1.4.8)

Can you try to remove this mapping, as my testing shows that when you remove this it works as expected.

**Comment by johnj24** [comment]
Excellent! That was the issue.
